src/Strategy/IfElse.php

```diff
@@ -0,0 +1,55 @@
+<?php
+namespace ScriptFUSION\Mapper\Strategy;
+
+use ScriptFUSION\Mapper\Mapping;
+
+/**
+ * Delegates to one expression or another depending on whether the specified condition loosely evaluates to true.
+ */
+class IfElse extends Delegate
+{
+    /** @var Strategy|Mapping|array|mixed */
+    private $if;
+
+    /** @var Strategy|Mapping|array|mixed */
+    private $else;
+
+    /**
+     * Initializes this instance with the specified condition, the specified
+     * strategy or mapping to be resolved when condition is non-null and,
+     * optionally, the specified strategy or mapping to be resolved when
+     * condition is null.
+     *
+     * @param Strategy|Mapping|array|mixed $condition Condition.
+     * @param Strategy|Mapping|array|mixed $if Primary expression.
+     * @param Strategy|Mapping|array|mixed|null $else Optional. Fallback expression.
+     */
+    public function __construct($condition, $if, $else = null)
+    {
+        parent::__construct($condition);
+
+        $this->if = $if;
+        $this->else = $else;
+    }
+
+    /**
+     * Resolves the stored strategy or mapping when the stored condition
+     * resolves to a non-null value, otherwise returns the stored default
```

Comment:
I don't understand this documentation. Why would we care if the resolved value was non-null? Aren't we interested in whether it is `true` or `false`?